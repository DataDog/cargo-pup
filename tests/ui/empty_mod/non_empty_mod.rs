//@compile-flags: --crate-name test
//@compile-flags: --crate-type lib

// This module simulates a mod.rs file with forbidden items
mod non_empty {
    // This is a re-export which should be allowed in mod.rs
    pub use std::collections::HashMap;
    
    // This struct definition should trigger the lint
    pub struct ForbiddenStruct { //~ ERROR: Item ForbiddenStruct disallowed in mod.rs due to empty-module policy
        pub field: String,
    }
    
    // This enum definition should trigger the lint
    pub enum ForbiddenEnum { //~ ERROR: Item ForbiddenEnum disallowed in mod.rs due to empty-module policy
        VariantA,
        VariantB,
    }
    
    // This trait definition should trigger the lint
    pub trait ForbiddenTrait { //~ ERROR: Item ForbiddenTrait disallowed in mod.rs due to empty-module policy
        fn method(&self);
    }
    
    // This struct impl should trigger the lint
    impl ForbiddenStruct { //~ ERROR: Item ForbiddenStruct disallowed in mod.rs due to empty-module policy
        pub fn new(value: String) -> Self {
            Self { field: value }
        }
    }
}