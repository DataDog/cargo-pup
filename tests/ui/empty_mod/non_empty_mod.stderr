error: Item ForbiddenStruct disallowed in mod.rs due to empty-module policy
  --> tests/ui/empty_mod/non_empty_mod.rs:10:5
   |
LL |     pub struct ForbiddenStruct {
   |     ^
   |
   = help: Remove this definition from the module.
   = note: Applied by cargo-pup rule 'empty_mod_test'.
   = note: `-D empty-mod` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(empty_mod)]`

error: Item ForbiddenEnum disallowed in mod.rs due to empty-module policy
  --> tests/ui/empty_mod/non_empty_mod.rs:15:5
   |
LL |     pub enum ForbiddenEnum {
   |     ^
   |
   = help: Remove this definition from the module.
   = note: Applied by cargo-pup rule 'empty_mod_test'.

error: Item ForbiddenTrait disallowed in mod.rs due to empty-module policy
  --> tests/ui/empty_mod/non_empty_mod.rs:21:5
   |
LL |     pub trait ForbiddenTrait {
   |     ^
   |
   = help: Remove this definition from the module.
   = note: Applied by cargo-pup rule 'empty_mod_test'.

error: Item ForbiddenStruct disallowed in mod.rs due to empty-module policy
  --> tests/ui/empty_mod/non_empty_mod.rs:26:5
   |
LL | /     impl ForbiddenStruct {
LL | |         pub fn new(value: String) -> Self {
LL | |             Self { field: value }
LL | |         }
LL | |     }
   | |_____^
   |
   = help: Remove this implementation from the module.
   = note: Applied by cargo-pup rule 'empty_mod_test'.

error: aborting due to 4 previous errors

